docs/user-guide.md

```diff
@@ -217,7 +259,86 @@ TODO
 from graphframes.examples import Graphs
 g = Graphs(sqlContext).friends()  # Get example graph
 
-TODO
+# Search for pairs of vertices with edges in both directions between them.
+motifs = g.find("(a)-[e]->(b); (b)-[e2]->(a)")
+motifs.show()
+
+# More complex queries can be expressed by applying filters.
+motifs.filter("e.relationship = 'follow' AND e2.relationship = 'follow'").show()
+{% endhighlight %}
+</div>
+
+</div>
+
+Many motif queries are stateless and simple to express, as in the examples above.
+More complex queries, such as those with state carried along a path in the motif,
+can likewise be expressed by combining GraphFrame motif finding with filters on the result.
+
+For example, suppose one wishes to identify a chain of 4 vertices with some property defined
+by a sequence of functions.  That is, among chains of 4 vertices `a->b->c->d`, identify the subset
+of chains matching this complex filter:
+
+* Initialize state on path.
+* Update state based on vertex `a`.
+* Update state based on vertex `b`.
+* Etc. for `c` and `d`.
+* If final state matches some condition, then the chain is accepted by the filter.
+
+The below code snippets demonstrate this process, where we identify chains of 4 vertices
+such that at least 2 of the 3 edges are "friend" relationships.
+The first query shows how to do this in a stateless way, and the second query shows how to do
+this with state, where the state is the current count of "friend" edges.
+
+<div class="codetabs">
+
+<div data-lang="scala"  markdown="1">
+{% highlight scala %}
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{col, when}
+import org.graphframes.examples
+
+val g = examples.Graphs.friends  // get example graph
+
+// Query on sequence, but without state
+val chain4 = g.find("(a)-[ab]->(b); (b)-[bc]->(c); (c)-[cd]->(d)")
+val chainWith2Friends = chain4.where(
+  Seq("ab", "bc", "cd")
```

Comment:
Maybe we need a better example or a precise definition for stateful motifs.

* For example, consider someone wants to plan a trip visiting several cities. He needs to spend some money between cities (on edges), and he could get some new money in cities. He wants to query all possible paths such that he won't run out of money at any location. This could be a stateful query, but I don't know how to express it clearly. It could be `var money = 10; (a where money > 0 | money += a.money)-[e where money > 0 | moeny -= e.cost]->(b where money > 0 | money += b.money)`.


* Another example is like BFS, I want to find paths that only visit each city once. So I need to mark cities as visited (state). The query might like `(a | a.state = "visited)->[e]-> (b where b.state != "visited" | b.state = "visited")`.

Just two examples, it is not very clear what we mean by "stateful motif".